plasmapy/formulary/parameters.py

```diff
@@ -537,16 +544,144 @@ def ion_sound_speed(
+       | ``"nrl"``                 | .. math::                                     |
+       |                           |    1                                          |
+       +--------------+------------+---------------+---------------+---------------+
+
+    Examples
+    --------
+    >>> thermal_speed_coefficients(method="most_probable", ndim=3)
+    1.414213...
+    """
+    _coefficients = {
+        (1, "most_probable"): 0,
+        (2, "most_probable"): 1,
+        (3, "most_probable"): np.sqrt(2),
+        (1, "rms"): 1,
+        (2, "rms"): np.sqrt(2),
+        (3, "rms"): np.sqrt(3),
+        (1, "mean_magnitude"): np.sqrt(2 / np.pi),
+        (2, "mean_magnitude"): np.sqrt(np.pi / 2),
+        (3, "mean_magnitude"): np.sqrt(8 / np.pi),
+        (1, "nrl"): 1,
+        (2, "nrl"): 1,
+        (3, "nrl"): 1,
+    }
+
+    try:
+        coeff = _coefficients[(ndim, method)]
+    except KeyError:
+        raise ValueError(
+            f"Value for (ndim, method) pair not valid, got '({ndim}, {method})'."
+        )
+
+    return coeff
+
+
+@preserve_signature
+@njit
+def thermal_speed_lite(
+    T: numbers.Real, mass: numbers.Real, coeff: numbers.Real
+) -> numbers.Real:
+    r"""
+    The "Lite-Function" version of `~plasmapy.formulary.parameters.thermal_speed`.
+    Performs the same thermal speed calculations as
+    `~plasmapy.formulary.parameters.thermal_speed`, but is intended for
+    computational use and, thus, has data conditioning safeguards removed.
+
+    .. math::
+        v_{th} = C_o \sqrt{\frac{k_B T}{m}}
+
+    where :math:`T` is the temperature associated with the distribution,
+    :math:`m` is the particle's mass, and :math:`C_o` is a constant of
+    proportionality determined by the method in which :math:`v_{th}` is
+    calculated and the dimensionality of the system (1D, 2D, 3D).  For further
+    details see the :ref:`Notes <thermal-speed-notes>` section in the
+    `~plasmapy.formulary.parameters.thermal_speed` documentation.
+
+    Parameters
+    ----------
+    T : `~numbers.Real`
+        The temperature of the particle distribution, in units of kelvin.
+
+    mass : `~numbers.Real`
+        Mass of the particle in kg.
+
+    coeff : `~numbers.Real`
+        The coefficient :math:`C_o` associated with the method used for
+        calculating the thermal speed, see :ref:`Notes <thermal-speed-notes>`
+        section in the `~plasmapy.formulary.parameters.thermal_speed`
+        documentation.
+
+    Returns
+    -------
+    vth : `~numbers.Real`
+        Thermal speed of the Maxwellian distribution in units of m/s.
+
+    Examples
+    --------
+    >>> from plasmapy.particles import Particle
+    >>> mass = Particle("p").mass.value
+    >>> coeff = thermal_speed_coefficients(method="most_probable", ndim=3)
+    >>> thermal_speed_lite(T=1e6, mass=mass, coeff=coeff)
+    128486...
+    """
+    return coeff * np.sqrt(k_B_si_unitless * T / mass)
```

Comment:
Note:  Since `thermal_speed_lite` is decorated by `numba.njit`, the code that is actually covered by tests is the c-compiled version and thus `coverage` does not recognize this gets hit during tests.

For now I'm updating the tests such that the following is essentially done...

```python
lite = thermal_speed_lite(**kwargs)
pylite = thermal_speed_lite.py_func(**kwargs)
assert lite == pylite
```

The call to `thermal_speed_lite.py_func` calls the original python function and, thus, is calculated in the coverage report.  An alternative to this is to add a scheme to the `excluded_lines` in the `[coverage:report]` settings of `setup.cfg` such that anything decorated with `jit` or `njit` is ignored.